adagrams/game.py

```diff
@@ -1,11 +1,89 @@
+import random
+# module will let us randomly choose characters in specified sample range
+
 def draw_letters():
-    pass
+    """
+    Build a hand of 10 letters for the user. Returns an array of strings, one letter long.
+    The num of strings included in the array 
+    cannot be more than the num alloted in the letter table
+    """
+    LETTER_POOL = {
+        'A': 9, 'B': 2, 'C': 2, 'D': 4, 'E': 12, 'F': 2, 'G': 3, 'H': 2, 'I': 9, 'J': 1,
+        'K': 1, 'L': 4, 'M': 2, 'N': 6, 'O': 8, 'P': 2, 'Q': 1, 'R': 6, 'S': 4, 'T': 6, 'U': 4,
+        'V': 2, 'W': 2, 'X': 1, 'Y': 2, 'Z': 1
+    }
+    letter_to_draw = list(LETTER_POOL.keys())
+    letters_drawn = []
+    letter_count = {}
+    for letter in random.sample(letter_to_draw, 10):
+        if letter not in letters_drawn:
+            letters_drawn.append(letter)
+            letter_count[letter] = 1
+        elif letter_count[letter] < LETTER_POOL[letter]:
+            letters_drawn.append(letter)
+            letter_count[letter] += 1
+
+    return letters_drawn
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    """
+    If letters in user's word are not in letter_bank, or if user uses more letters
+    than in hand, the function returns False. 
+    Function returns False if user enters empty word. Else, it returns True.
+    """
+    for letter in word:
+        if letter not in letter_bank or word.count(letter) > letter_bank.count(letter):
+            is_valid = False
+        else:
+            is_valid = True
+
+    return is_valid
```

Comment:
Great use of a guard clause here. We can make the code even neater by getting rid of the `else` clause (aka the dangly bit). To do so, we can take advantage of using the `is_valid` flag to describe the single scenario where `is_valid` would be false in the guard clause. Otherwise, the function will just return True.  Like so:

```
    is_valid  = True 
    for letter in word:
        if letter not in letter_bank or word.count(letter) > letter_bank.count(letter):
            is_valid = False

    return is_valid
```

```suggestion

    is_valid  = True 
    for letter in word:
        if letter not in letter_bank or word.count(letter) > letter_bank.count(letter):
            is_valid = False

    return is_valid
```